File: csharp/FantasyBattle/BasicItem.cs
namespace FantasyBattle
{
    public class BasicItem : Item
    {
        public BasicItem(string name, int baseDamage, float damageModifier)
        {
            BaseDamage = baseDamage;
            DamageModifier = damageModifier;
        }

        public int BaseDamage { get; }
        public float DamageModifier { get; }
    }
}

File: typescript/src/BasicItem.ts
import {Item} from './Item';

export class BasicItem implements Item {

    public constructor(private name: string,
                       private _baseDamage: number,
                       private _damageModifier: number) {
    }

    public get baseDamage(): number {
        return this._baseDamage;
    }

    public get damageModifier(): number {
        return this._damageModifier;
    }
}
